spec/check_please/comparison_spec.rb

```diff
@@ -147,6 +147,135 @@ def invoke!(ref, can, flags = {})
     end
   end
 
+  describe "comparing arrays by keys" do
```

Comment:
If I understand correctly, match by key means array's are matched without regard to order?

Reply:
Yeah, I really need to figure out how and where to document this for the end user. (Fortunately, at the moment, it me.) "Match by key" means that if you have an array of hashes, you can specify a single hash attribute to use as the identifier for that hash, and we'll try to match up the hashes using that identifier, rather than by position in the array.  Below that point, the comparison algorithm proceeds as usual, which is why the path expression syntax supports multiple key expressions...